x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts

```diff
@@ -40,66 +45,64 @@ export async function getServiceMapServiceNodeInfo({
   const filter: ESFilter[] = [
     { range: rangeFilter(start, end) },
     { term: { [SERVICE_NAME]: serviceName } },
-    ...(environment ? [{ term: { [SERVICE_ENVIRONMENT]: environment } }] : []),
   ];
 
+  const environmentFilter = getEnvironmentUiFilterES(environment);
+
+  if (environmentFilter) {
+    filter.push(environmentFilter);
+  }
+
   const minutes = Math.abs((end - start) / (1000 * 60));
 
   const taskParams = {
+    serviceName,
     setup,
     minutes,
     filter,
   };
 
   const [
-    errorMetrics,
-    transactionMetrics,
-    cpuMetrics,
-    memoryMetrics,
+    errorStats,
+    transactionStats,
+    cpuStats,
+    memoryStats,
   ] = await Promise.all([
-    getErrorMetrics(taskParams),
-    getTransactionMetrics(taskParams),
-    getCpuMetrics(taskParams),
-    getMemoryMetrics(taskParams),
+    getErrorStats(taskParams),
+    getTransactionStats(taskParams),
+    getCpuStats(taskParams),
+    getMemoryStats(taskParams),
   ]);
 
   return {
-    ...errorMetrics,
-    ...transactionMetrics,
-    ...cpuMetrics,
-    ...memoryMetrics,
+    ...errorStats,
+    ...transactionStats,
+    ...cpuStats,
+    ...memoryStats,
   };
 }
 
-async function getErrorMetrics({ setup, minutes, filter }: TaskParameters) {
-  const { client, indices } = setup;
-
-  const response = await client.search({
-    index: indices['apm_oss.errorIndices'],
-    body: {
-      size: 0,
-      query: {
-        bool: {
-          filter: filter.concat({
-            term: {
-              [PROCESSOR_EVENT]: 'error',
-            },
-          }),
-        },
-      },
-      track_total_hits: true,
+async function getErrorStats({
+  filter,
+  serviceName,
+  setup,
+}: {
+  filter: ESFilter[];
+  serviceName: string;
+  setup: Options['setup'];
+}) {
+  const { average, noHits } = await getErrorRate({
+    serviceName,
+    setup: {
+      ...setup,
+      uiFiltersES: filter,
```

Comment:
@dgieselaar This is one of the issues I see with `setup`. When we start overriding properties like this it has become too opaque. 
As a rule of thumb we should never do this - if we need to I think it's better to pass each prop in individually (not as `setup`)